banks/serializers.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from .models import Bank, Branch, Account, Transaction, TransactionType
+
+
+class BankSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Bank
+        fields = '__all__'
+
+
+class AccountSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Account
+        fields = '__all__'
+
+
+class BranchSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Branch
+        fields = '__all__'
+
+
+class TransactionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Transaction
+        fields = '__all__'
+
+    def validate_amount(self, value):
+        account_balance = Account.objects.get(id=self.initial_data['account']).balance
```

Comment:
check if you can get validated data here otherwise check [here](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation)